docker/data-science/notebooks/ch08/1.signals-boosting.ipynb

```diff
@@ -518,22 +513,22 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 19,
    "metadata": {},
    "outputs": [
     {
      "name": "stdout",
      "output_type": "stream",
      "text": [
       "Boost Documents:\n",
-      "[{'query': 'star wars', 'doc': '45626176', 'boost': 5000}, {'query': 'star wars', 'doc': '024543742180', 'boost': 1490}, {'query': 'star wars', 'doc': '400032015667', 'boost': 186}, {'query': 'star wars', 'doc': '024543742074', 'boost': 127}, {'query': 'star wars', 'doc': '024543559856', 'boost': 117}, {'query': 'star wars', 'doc': '014633169546', 'boost': 107}, {'query': 'star wars', 'doc': '024543742098', 'boost': 81}, {'query': 'star wars', 'doc': '024543781875', 'boost': 67}, {'query': 'star wars', 'doc': '023272342654', 'boost': 39}, {'query': 'star wars', 'doc': '014633169522', 'boost': 32}, {'query': 'star wars', 'doc': '024543560067', 'boost': 30}, {'query': 'star wars', 'doc': '883929154012', 'boost': 24}, {'query': 'star wars', 'doc': '023272342630', 'boost': 20}, {'query': 'star wars', 'doc': '024543263739', 'boost': 18}, {'query': 'star wars', 'doc': '708431390614', 'boost': 18}, {'query': 'star wars', 'doc': '886973561621', 'boost': 17}, {'query': 'star wars', 'doc': '023272341381', 'boost': 16}, {'query': 'star wars', 'doc': '030206742121', 'boost': 14}, {'query': 'star wars', 'doc': '885370332889', 'boost': 12}, {'query': 'star wars', 'doc': '883929106172', 'boost': 8}]\n",
+      "[{'query': 'star wars', 'doc': '024543742180', 'boost': 1490}, {'query': 'star wars', 'doc': '400032015667', 'boost': 186}, {'query': 'star wars', 'doc': '024543742074', 'boost': 127}, {'query': 'star wars', 'doc': '024543559856', 'boost': 117}, {'query': 'star wars', 'doc': '014633169546', 'boost': 107}, {'query': 'star wars', 'doc': '024543742098', 'boost': 81}, {'query': 'star wars', 'doc': '024543781875', 'boost': 67}, {'query': 'star wars', 'doc': '023272342654', 'boost': 39}, {'query': 'star wars', 'doc': '014633169522', 'boost': 32}, {'query': 'star wars', 'doc': '024543560067', 'boost': 30}, {'query': 'star wars', 'doc': '883929154012', 'boost': 24}, {'query': 'star wars', 'doc': '023272342630', 'boost': 20}, {'query': 'star wars', 'doc': '708431390614', 'boost': 18}, {'query': 'star wars', 'doc': '024543263739', 'boost': 18}, {'query': 'star wars', 'doc': '886973561621', 'boost': 17}, {'query': 'star wars', 'doc': '023272341381', 'boost': 16}, {'query': 'star wars', 'doc': '030206742121', 'boost': 14}, {'query': 'star wars', 'doc': '885370332889', 'boost': 12}, {'query': 'star wars', 'doc': '883929106172', 'boost': 8}, {'query': 'star wars', 'doc': '886111287055', 'boost': 8}]\n",
       "\n",
       "Boost Query: \n",
```

Comment:
Note: the "5000 spam signals" issue is present here again. In this case, it looks like the new changes are correct and fixing the old changes. HOWEVER, there's non-deterministic sorting of the boosts, which are making the output vary. Possibly we need a secondary sort on the upc?